include/zephyr/linker/common-rom/common-rom-kernel-device-apis.ld

```diff
@@ -0,0 +1,7 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+	SECTION_PROLOGUE(device_apis,,)
+	{
+		CREATE_DEVICE_API_LIST(location)
+		CREATE_DEVICE_API_LIST(uart)
```

Comment:
This won't work with out-of-tree API definitions.

Reply:
It is not strictly necessary unless we want to be able to iterate through all devices of a specific API type, we can sort them by name, since the section name is z_device_api_<api_type>_ ... but can we determine start and stop address of them somehow?

Reply:
I removed the creation of iterable sections in favor of simply adding all of the devices to a section in ROM, this PR now works both for in-tree and out-of-tree APIs, at the cost of not being able to iterate by api_type. This can be added in the future if we discover a way to do that properly.